### ISSUE TITLE
[BUG] Valve Index controllers permanently rumbling

### ISSUE THREAD

**Issue by Corben78** [ValveSoftware/SteamVR-for-Linux#268]
**Description**
With Steam VR Beta 1.8.18 I realized that the Valve Index controllers start permanently rumbling after turning them on. This did not happen with 1.7.x before, but still happened with 1.8.19 and still happens with 1.8.20. After starting them up, they shortly rumble, pause, and then permanently rumble. If I hold them a big tighter though, they stop rumbling. As soon as I hold them more loosely again, they continue the rumble.
Also the Vive Pro wands don't rumble permanently.

1. Start Steam VR
2. Turn on Valve Index controllers
3. Feel the rumble.

**Expected behavior**
Valve Index controllers should only rumble when initiated by an application.

**System Information (please complete the following information):**
- Distribution: Ubuntu 18.04 LTS
- SteamVR version: 1.8.20 beta
- Steam client version: 1573176642 Build Nov 7 2019 at 23:57:46
- Opted into Steam client beta?: Yes
- Graphics driver version:  440.26
- Gist for SteamVR System Information: [gist](https://gist.github.com/Corben78/448924b0259c190cf91781bb13ad9281)
- Gist for Steam Client System Information: [gist](https://gist.github.com/Corben78/2551e74077522366cb0fd31b7c0cbd25)

**Comment by Zamundaaa** [comment]
I had this happen twice now, too. When it happened I fixed it by power cycling the headset, it didn't rumble on the next start of SteamVR.

**Comment by StefanTT** [comment]
Some games seem to cause this. Sometimes it helps to power cycle the HMD as Zamundaaa said.
Interesting fact: the rumbling is persistent across games. When a game triggers the behaviour then it cannot always be fixed by quitting the game.

A free game that causes the rumbling for me is "In Your Face TD".

**Comment by Corben78** [comment]
For me this happens without any game running though. Just SteamVR, which I have without SteamVR Home running.

**Comment by Zamundaaa** [comment]
Same for me, the rumbling just starts after SteamVR starts, no application required at all.

**Comment by StefanTT** [comment]
Yes, this is the hard case. Fortunately it seldom happens to me.

I usually can get rid of it by resetting the headset via SteamVR > Devices > Restart headset
and/or opening the controller bindings via SteamVR > Devices > Controller settings
If this is not enough sometimes killing all Steam / -VR processes helps

**Comment by thomasvring** [comment]
I've found that opening the debug commands window fixes it (vrmonitor > settings button, developer > debug commands). You have to open the debug commands window, close it after it's fully loaded, then open it again. This time it will crash, and the rumbling stops! You have to do it after starting a game though, if you do it in SteamVR Home and then open a game, the rumbling starts again and you have to do the fix again.

**Comment by StefanTT** [comment]
Unfortunately the debug commands does not fix it for me. At least it did not help now. Killing all steam processes helped again. If anybody needs it, here is my crude "kill steam" script:
```
#!/bin/bash
ps aux|grep -i steam|grep -v killall|grep -v grep|awk '{print $2}'|xargs kill
sleep 3 
ps aux|grep -i steam|grep -v killall|grep -v grep|awk '{print $2}'|xargs kill -9
```

**Comment by thomasvring** [comment]
I used to restart Steam to fix it, but that stopped working for me at some point. Then I thought, "maybe there's a debug option to turn rumble off for now". Noticed rumble stopped for a sec when I opened the debug commands window. Tried opening the window again just to check and then the rumbling stopped.

 For me the rumbling starts again when starting a game, it seems that doesn't happen for you and others. There must be something more going on here if different things affect the constant rumbling.

**Comment by Mastergatto** [comment]
Also, the controllers always rumble when a game is being launched directly, skipping SteamVR.

**Comment by Zamundaaa** [comment]
I can confirm that. The rumbling then does not stop if I close the game, I have to quit, or more accurately kill (because of #209) SteamVR. Here's the system report from the next start:
https://gist.github.com/Zamundaaa/0ecfdda149fa20d0fdf744497ff737b9
If the diagnostics added in 1.9.5 are not in the system report then please let us know.

**Comment by Corben78** [comment]
Do you guys have the launch option STEAMVR_WEBHELPER=0 for SteamVR set?
I had, and as soon as I removed it, the perma-rumble was gone.

**Comment by thomasvring** [comment]
I do not have that launch option set.

Weird but pleasant; the last two times I've now started SteamVR and a VR game, no constant rumbling. Nothing changed on my side.

**Comment by Matombo** [comment]
Found a reproducable workaround when it happens:
1. turn of the controllers
2. close steamvr
3. close steam
4. turn on the controlers (while neither steam or steam vr is running)
5. start steam again
6. start steam vr again

**Comment by Zamundaaa** [comment]
Had it happen again just now because I started SteamVR by turning on the controllers (seems to be reproducable this way; starting directly from Steam works without constant rumbling). At least that now (partially) works again :)
https://gist.github.com/Zamundaaa/f1990b8dfcb5b19bc8aebe5f647afdb0

**Comment by Zamundaaa** [comment]
Starting a VR app without SteamVR already running still consistently reproduces this bug with SteamVR 1.10.16 and Steam beta version 1582070848

**Comment by kherchel** [comment]
Issue still happening with SteamVR 1.10.30 and Steam Beta from Mar 9 2020, at 22:39:56.
Using Mesa 19.3.4 (LLVM 9.0.1) on an up-to-date Arch Linux.
Gist for SteamVR System Information: https://gist.github.com/kherchel/d6c22295d29dab434ce1c7e030a4afc5

What I've noticed is that the harder I grip the controllers, the less they vibrate - when gripping at full force they don't vibrate at all actually. They also respond to some events like running games, and stop vibrating for a moment.

**Comment by langurmonkey** [comment]
My application shows this behaviour with the Valve Index. Works fine with other headsets. The rumbling happens when polling the game state to get the controller axis' values in `VRSystem.GetControllerState(index, state)`. When I debug this step-by-step, the controller registers a small rumble every time this function is called. I have no idea why this happens.
I launch my app directly, not from Steam, even though it uses OpenVR and obviously connects to the SteamVR runtime (which must be running).
